### ISSUE TITLE
Snap crashes with Segmentation fault (core dumped)

### ISSUE THREAD

**Issue by kspearrin** [electron-userland/electron-builder#2981]
* **Version**: ^20.15.1

* **Target**: --linux --x64

We have been publishing our snap in the past without issue. We recently updated to electron 2.0.2 and builder 20.15.1 and now we get the following crash when trying to run a built snap. This seems to happen on Ubuntu 16, 17, and 18.

```
$ snap run myapp
Gtk-Message: Failed to load module "canberra-gtk-module"
Gtk-Message: Failed to load module "canberra-gtk-module"
```
The Gtk errors are normal. A long pause of ~15 seconds happens. Then I see the app window appear for a split second and then.

```
Segmentation fault (core dumped)
```

[This issue](https://github.com/electron-userland/electron-builder/issues/2808) seems related, but was said to have been fixed.

Reference app: https://github.com/bitwarden/desktop

Building AppImage works and runs fine.

**Comment by dimitarnestorov** [comment]
@kspearrin Have you managed to fix it? I'm experiencing the same issue.

**Comment by develar** [comment]
@dimitarnestorov Do you build on Linux?

**Comment by dimitarnestorov** [comment]
@develar No (I use the template, on macOS). But I will try that now

**Comment by develar** [comment]
Ok... Please set env `_REMOTE_BUILD=true` does it help? If not, please  "Is option `useTemplateApp` (under `snap`) help you?" (only on Linux or macOS with Docker).

**Comment by develar** [comment]
If your app has native deps and no prebuild, you can build only on Linux.

**Comment by kspearrin** [comment]
@dimitarnestorov @develar  Have not found a fix yet. We build on Ubuntu.

**Comment by dimitarnestorov** [comment]
Tried `_REMOTE_BUILD=true`, didn't work.
Tried `"useTemplateApp": true` on Ubuntu, didn't work.
Tried `"useTemplateApp": false` on Ubuntu, 2 hours later, didn't work.

**Comment by develar** [comment]
Do you have native deps?

**Comment by kspearrin** [comment]
Yes, see the repo mentioned in the original issue: https://github.com/bitwarden/desktop

**Comment by dimitarnestorov** [comment]
> Do you have native deps?
No.

**Comment by dimitarnestorov** [comment]
I'm using the package `@sentry/electron`@0.5.4 for error tracking. Calling it's init method is the cause of this error on my app.

**Comment by develar** [comment]
@dimitarnestorov So, it is app specific issue?

**Comment by dimitarnestorov** [comment]
@develar Yes: https://github.com/getsentry/sentry-electron/issues/98

**Comment by kspearrin** [comment]
@develar  Any reason why this was closed? My original issue still remains. I am not using the referenced sentry-electron package.

**Comment by develar** [comment]
@kspearrin "reply needed" was added to get answer from you. And now you have answered :)

**Comment by dimitarnestorov** [comment]
I just tried building electron-quick-start with your snap configuration and it worked
![image](https://user-images.githubusercontent.com/8790386/43015893-90b68fe0-8c59-11e8-9b83-2933621fc347.png)

You should check out your runtime code for something similar to my issue.

**Comment by develar** [comment]
@kspearrin please check your app works as unpacked / AppImage.

**Comment by kspearrin** [comment]
@develar Yes, it does. We are distributing it already as an AppImage under that version.

**Comment by develar** [comment]
@kspearrin have you tried useTemplateApp: false?

**Comment by kspearrin** [comment]
@develar Is there a way for me to get more feedback about what's crashing the app here other than just "Segmentation fault (core dumped)" . It doesn't seem to be very helpful.

**Comment by kspearrin** [comment]
@develar Updated to electron 2.0.5 and builder ^20.25.0 , but it made no difference. Same problem still. Can this be reopened please?

**Comment by kspearrin** [comment]
@develar I've spent today trying to debug this issue more.

I placed `console.log` message all throughout the app startup process and I can see that the main process executes without issue. The BrowserWindow is created, index.html loaded, and window.show() without any error.

Then `Segmentation fault (core dumped)` error happens.

Line 1 of the renderer process window's JavaScript application never executes (I have a console.log there to test).
